managed/src/main/java/com/yugabyte/yw/controllers/CloudProviderController.java

```diff
@@ -172,6 +176,11 @@ public Result create(UUID customerUUID) throws IOException {
       if (!config.isEmpty()) {
         switch (provider.code) {
           case "aws":
+            CloudAPI cloudAPI = cloudAPIFactory.get(provider.code);
```

Comment:
I'm not sure we want to make some kind of blocking call to AWS in the main API request thread. IMO this should be done as an asynchronous subtask for creating the cloud provider which fails the task if the credentials are invalid.

If it is moved to an async subtask, I would also argue that we should stick with using boto for now since we have not flushed out how exactly we want to use the Java layer to interact with cloud providers yet cc: @sb-yb on if you agree with this or not. My concern is that you implement it for AWS at the Java layer and then have to either implement CloudAPI for the other cloud provider types (GCP, Azure, etc) or that you have the AWS solution in Java and all the other solutions in python.

Since we haven't had time to plan out how we want to properly interact with cloud providers (i.e. whether we want to move everything to the Java layer or not), it's best to stick with the current pattern of using python for now (as discussed in the Slack channel for the approach to take for pausing a universe).

Reply:
So about async part - I think we are not using async handler in our controller. So no matter what we do behind the scene, the incoming API call will block. This is a general issue that I mentioned to @yb-andrew during one of his reviews. 
But we decided not to do any optimization at this point. It is really not-worth for the qps that we serve in platform. 

The later problem is something I have to solve for missing instance types GCP and Azure anyway. So they can implement after that.